### DISCUSSION TITLE
How to elegantly store a list of scripts?

### DISCUSSION THREAD

**Discussion by jespertheend** [denoland/deno#13004] [Q&A] [answered]
Using npm we can run `npm run <script>` to run a specific command from `package.json`, I'd like to do something similar with Deno.
I know you can get a long way with `deno install`, but I'm looking for a way to include a list of commands with my repository. That way new team members can easily see what the available commands are and run them without having to guess the required permissions and flags.
Is there a good way to do this? For now I have added a bunch of tasks to `.vscode/tasks.json`, but I was hoping for something more IDE agnostic.

**Comment by bartlomieju** [comment, marked as answer]
See https://github.com/denoland/deno/issues/12764

**Reply by jespertheend** [reply]
Ah yes this is exactly what I'm looking for, thanks!

**Reply by lucacasonato** [reply]
@jespertheend A great way to do this already is using a Makefile:

```makefile
# Makefile

run:
  deno run --allow-read --allow-net main.ts

lint:
  deno lint
```

You can then run these with `make run` or `make lint`.

**Reply by jespertheend** [reply]
Ah thanks! Maybe it's time I finally learn how makefiles work then. Though I'd prefer a built-in solution for deno, this seems like a good alternative for the time being.

**Comment by abflow** [comment]
You may also look at the denon module, which can store scripts in a scripts.json file with various options like permissions, etc., in the same way as the proposal above.

You can then run a script with `denon [script name]`
https://github.com/denosaurs/denon#json-config-scriptsjson-template

**Comment by jespertheend** [comment]
Using a shebang can also get a long way:
`#!/usr/bin/env -S deno run --unstable --allow-read --allow-write`
You won't have all scripts in a single list, but if you put them all in roughly the same directory it can be pretty organised.

I saw this in [tools/format.js](https://github.com/denoland/deno/blob/9825c876b466cf69a8ddd1646ba7fedecd54ba51/tools/format.js#L1) and I kind of like it.
